internal/provider/provider.go

```diff
@@ -31,12 +37,20 @@ type (
 
 // Ensure AtlasProvider satisfies various provider interfaces.
 var (
-	_ provider.Provider             = &AtlasProvider{}
-	_ provider.ProviderWithMetadata = &AtlasProvider{}
+	_ provider.Provider                   = &AtlasProvider{}
+	_ provider.ProviderWithMetadata       = &AtlasProvider{}
+	_ provider.ProviderWithValidateConfig = &AtlasProvider{}
+)
+
+const (
+	// envNoUpdate when enabled it cancels checking for update
+	envNoUpdate = "ATLAS_NO_UPDATE_NOTIFIER"
+	vercheckURL = "https://vercheck.ariga.io"
+	versionFile = "~/.atlas/terraform-provider-release.json"
```

Comment:
I think with the TF provider it makes sense to store the version file adjacent to the provider binary. (It is stored somewhere in .terraform in the project dir, I believe)